demonstrations/tutorial_liealgebra.py

```diff
@@ -210,51 +215,73 @@
 
 generators = [1j * (X(0) @ X(1)), 1j * Z(0), 1j * Z(1)]
 
-dla = generators.copy()
+dla = qml.pauli.PauliVSpace(generators, dtype=complex) # collection of linearly independent basis vectors, automatically discards linearly dependent ones
 for i, op1 in enumerate(generators):
     for op2 in generators[i+1:]:
         res = qml.commutator(op1, op2)/2
         res = res.simplify() # ensures all products of scalars are executed
         print(f"[{op1}, {op2}] = {res}")
 
-        # add new operator to dla, normalize for convenience
-        if res.scalar != 0. and -1*res not in dla and res not in dla:
-
+        if res.scalar != 0. and dla.is_independent(res.pauli_rep):
+            # Note that the previous is_independent check is just for pedagocical purposes
+            # as dla.add already checks linear independence below
             print(f"Appending {res}")
-            dla.append(res)
+            dla.add(res)
 
 ##############################################################################
 # We obtain two new operators :math:`iY_0 X_1` and :math:`iX_0 Y_1` and append the list of operators of the DLA.
 # We then continue with depth-1 nested commutators ("nested" as :math:`iY_0 X_1 \propto [iX_0 X_1, iZ_0]`).
 
-for i, op1 in enumerate(dla.copy()):
-    for op2 in dla.copy()[i+1:]:
+for i, op1 in enumerate(dla.basis.copy()):
+    for op2 in dla.basis.copy()[i+1:]:
         res = qml.commutator(op1, op2)/2
         res = res.simplify()
         print(f"[{op1}, {op2}] = {res}")
 
-        # add new operator to dla, normalize for convenience
-        if res.scalar !=0. and -1*res not in dla and res not in dla:
-
+        if res.scalar != 0. and dla.is_independent(res.pauli_rep):
             print(f"Appending {res}")
-            dla.append(res)
+            dla.add(res)
 
 ##############################################################################
 # The only new operator here is :math:`iY_0 Y_1`, which we add to the list of the DLA.
 # We could continue this process with a second nesting layer but will find that no new operators are added past this point.
 # We finally end up with the DLA :math:`\{X_0 X_1, Z_0, Z_1, iY_0 X_1, iX_0 Y_1, iY_0 Y_1\}`
 
-for op in dla:      
+for op in dla.basis:      
     print(op)
 
 ##############################################################################
 # Curiously, even though both :math:`iZ_0` and :math:`iZ_1` are in the DLA, :math:`iZ_0 Z_1` is not.
 # Hence, products of generators are not necessarily in the DLA.
 #
+# We have constructed the DLA by hand to showcase the process. We can use the PennyLane function :func:`~lie_closure` for convenience.
+# In that case, we omit the explicit use of the imgaginary factor.
+
+dla2 = qml.lie_closure([X(0) @ X(1), Z(0), Z(1)])
+dla2
+
```

Comment:
```suggestion

###############################################################
```

This should solve the tenderization issue